app.py

```diff
@@ -1,80 +1,94 @@
-'''
+"""
 Flask Application
-'''
+"""
 from flask import Flask, jsonify, request
 from models import Experience, Education, Skill
 
 app = Flask(__name__)
 
 data = {
     "experience": [
-        Experience("Software Developer",
-                   "A Cool Company",
-                   "October 2022",
-                   "Present",
-                   "Writing Python Code",
-                   "example-logo.png")
+        Experience(
+            "Software Developer",
+            "A Cool Company",
+            "October 2022",
+            "Present",
+            "Writing Python Code",
+            "example-logo.png",
+        )
     ],
     "education": [
-        Education("Computer Science",
-                  "University of Tech",
-                  "September 2019",
-                  "July 2022",
-                  "80%",
-                  "example-logo.png")
+        Education(
+            "Computer Science",
+            "University of Tech",
+            "September 2019",
+            "July 2022",
+            "80%",
+            "example-logo.png",
+        )
     ],
-    "skill": [
-        Skill("Python",
-              "1-2 Years",
-              "example-logo.png")
-    ]
+    "skill": [Skill("Python", "1-2 Years", "example-logo.png")],
 }
 
 
-@app.route('/test')
+@app.route("/test")
 def hello_world():
-    '''
+    """
     Returns a JSON test message
-    '''
+    """
     return jsonify({"message": "Hello, World!"})
 
 
-@app.route('/resume/experience', methods=['GET', 'POST'])
+@app.route("/resume/experience", methods=["GET", "POST"])
 def experience():
-    '''
+    """
     Handle experience requests
-    '''
-    if request.method == 'GET':
-        return jsonify()
+    """
+    if request.method == "GET":
+        try:
+            index = request.args.get("index")
+            if index:
+                index = int(index)
+                if 0 <= index < len(data["experience"]):
+                    specific_experience = data["experience"][index]
+                    return jsonify(specific_experience)
+                else:
+                    return jsonify({"error": "Index out of range"}), 404
 
-    if request.method == 'POST':
+            return jsonify(data["experience"])
+
+        except (TypeError, ValueError):
+            return (jsonify({"error": "Invalid index"}),), 400
```

Comment:
Do you think we could be more specific in the error? For instance, we could check if the field `index` exists and if it doesn't, instruct the user to call this API endpoint by providing the `index` field. Also, that would help in avoiding to keep all of the logic inside a try-catch block which would be nice to avoid IMHO.

Reply:
I don't really get this.